### DISCUSSION TITLE
eclipse 2023-06 + devtoolset-11 (gcc 11.2.1): indexer fails with std::set<> and std::map<>

### DISCUSSION THREAD

**Discussion by net1701** [eclipse-cdt/cdt#477] [Q&A] [answered]
it seems that the combination indicated in title give origin to problems.

Even a very simple source file just declaring:

`std::map <int, int> m;`

will be marked as error by indexer, and consequently all methods of, and references to, the declared object.

The reported error is just: "Invalid template arguments".
How should I investigate more deeply?  I thought it could be that the output of `-E -P -v -dD` is slighlty different, but it seems really the same. 

It works fine if the compiler is the native compiler: this is an old centos7 image, gcc version 4.8.5.
Other STL classes and containers work fine, even with the updated compiler.

NOTE: I'm working with makefile based projects

**Comment by jonahgraham** [comment, marked as answer]
I think this discussion refers to the same problem as described in this issue https://github.com/eclipse-cdt/cdt/issues/438 - if it doesn't seem to be the same, please can you provide some more details?

**Reply by net1701** [reply]
I can confirm that setting -std=c++14, the problem vanishes - the issue is the same. The problem originated originally from a cmake-generated eclipse project, where I cannot find easilty where to set options, but then I have found out that it can be easly reproduced with a simple hello-world example. Thanks.

**Comment by jonahgraham** [comment]
Can you try the suggestion in #438:

> Before support for deduction guides landed it did not fully worked anyway for >= c++17 so if you do not use that set `--std=c++14` or earlier.
